src/services/drpcEvents/index.ts

```diff
@@ -241,4 +249,85 @@ export default class DrpcEvents {
       })
     }
   }
+
+  private async handleParentQuery(id: UUID, childQuery: QueryRow, params: SubmitQueryResponseRPCParams) {
+    const [parentQuery]: QueryRow[] = await this.db.get('query', { id })
+    if (!parentQuery) {
+      this.logger.warn('parent query not found %s', id)
+      return
+    }
+
+    // adding childs and parent emissions
+    const total: number = parseInt(childQuery.details.emissions) + parseInt(params.emissions || '0')
+    await this.db.update(
+      'query',
+      { id },
+      {
+        status: 'resolved',
+        details: {
+          ...parentQuery.details,
+          emissions: total.toString(),
+        },
+      }
+    )
+
+    const [connection] = await this.db.get('connection', { id: parentQuery.connection_id })
+    if (!connection.agent_connection_id) return this.handleError(childQuery, connection)
+
+    let rpcResponse: DrpcResponse
+    try {
+      const maybeResponse = await this.cloudagent.submitDrpcRequest(
+        connection.agent_connection_id,
+        'submit_query_response',
+        {
+          query: 'Scope 3 Carbon Consumption',
+          emissions: total.toString(),
+          queryIdForResponse: parentQuery.response_id as UUID,
+        }
+      )
+      this.logger.info('submitting DRPC request %j', maybeResponse)
+      if (!maybeResponse) {
+        this.logger.warn('no response')
+        return this.handleError(childQuery, connection)
+      }
+      rpcResponse = maybeResponse
+    } catch (err) {
+      this.logger.warn('DRPC has failed %j', err)
+      return this.handleError(childQuery, connection)
+    }
+    const { result, error, id: rpcId } = rpcResponse
+
+    this.logger.info('persisting query_rpc response %j', rpcResponse)
+    await this.db.insert('query_rpc', {
+      agent_rpc_id: rpcId,
+      query_id: parentQuery.id,
+      role: 'client', // am I still a client when I'm a 'responder'?
```

Comment:
yes this is correct. Remove the comment please though

Reply:
removed